code/datums/components/kaleidoscope.dm

```diff
@@ -0,0 +1,52 @@
+/datum/component/kaleidoscope
+	var/list/colours
+	var/mode = KALEDIOSCOPE_RANDOM_COLOUR
+	var/colour_priority = ADMIN_COLOUR_PRIORITY
+	var/delay = 1
+	var/cleanup = TRUE
+	var/_index = 1 // Curse you DM, and your 1-indexed lists
+
+/datum/component/kaleidoscope/Initialize()
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	paint()
+
+/datum/component/kaleidoscope/proc/paint()
+	if(QDELETED(src))
+		return
+	var/chosen
+	switch(mode)
+		if(KALEDIOSCOPE_ORDERED_LIST)
+			chosen = colours[_index]
+			if(_index >= length(colours))
+				_index = 1
```

Comment:
Is this not just `next_list_item()`?
https://github.com/tgstation/tgstation/blob/a89c5aee03f3660d31715f268b192270e90abeae/code/__HELPERS/_lists.dm#L283-L290